frontend/src/views/consumer/paymentstatus-modal/index.js

```diff
@@ -0,0 +1,111 @@
+      state = {
+        show: true,
+      };
+      getColor = statusCode => {
+        const { red, primary, lightGreen, purple } = this.props.theme.palette;
+        if (statusCode == 0) return lightGreen;
+        if (statusCode == 1) return red;
+        if (statusCode == 2) return primary;
+        if (statusCode == 3) return purple;
+        if (statusCode == 4) return red;
+        return red;
+      };
+      // @TODO: get translation for these text
+      getMeaning = statusCode => {
+        if (statusCode == 0) return 'Maksu vahvistettu';
+        if (statusCode == 1) return 'Payment not in system';
+        if (statusCode == 2) return 'Already paid';
+        if (statusCode == 3) return 'Bambora failure';
+        if (statusCode == 4) return 'Maksu epäonnistui';
+        return 'Unknown Error';
+      };
+      getMeaningLong = statusCode => {
+        if (statusCode == 0)
+          return `Kiitos ostostasi! Lähetämme liput antamaasi 
+        puhelinnumeroon tekstiviestillä.`;
+        // @TODO: Wording to handle this case?
+        if (statusCode == 1)
+          return `It seems like you have stumbled upon a weird error. Please let our engineer know what happened`;
+        if (statusCode == 2)
+          return `You already paid for this event. The order that you requested are therefore not fulfilled. You have not been charge anything.`;
+        if (statusCode == 3)
+          return 'Our third party payment system seemed to have a problem at the moment. Your order have not been fulfilled. We are sorry for this inconvenience.';
+        if (statusCode == 4)
+          return `Olemme pahoillamme!
+          Maksunvälityksessä tapahtui häiriö, eikä
+          maksun tekeminen onnistunut.
+          `;
+        return 'Some thing strange happened and we could not fulfill your order. Please try again.';
+      };
+      render() {
+        return (
+          <Route
+            path="/consumer/payment"
+            children={({ match, location }) => {
+              const query = parseQuery(location.search);
+              const meaning = match && this.getMeaning(query.status);
+              const longText = match && this.getMeaningLong(query.status);
+              const event = this.props.store.events.toJSON()[query.event_id];
+              return (
+                <Modal
+                  show={match && this.state.show}
+                  onClear={() => this.setState({ show: false })}
+                >
+                  <Content>
+                    <Typography
+                      type="title"
+                      color={match && this.getColor(query.status)}
+                    >
+                      {meaning}
+                    </Typography>
+                    <Typography style={{ whiteSpace: 'pre-line' }} type="body">
+                      {longText}
+                    </Typography>
+                    <br />
+                    {query.status == 4 &&
```

Comment:
With enumeration on status codes this would be more readable:
`query.status === Statuses.PAYMENT_FAILED`

Also remember to use `===` instead of `==`

Reply:
some of the other piece of code throw because of === so i revert back to == instead.

Reply:
status code will have enums value at later patch. now too much added work

Reply:
👍